src/blocks.h

```diff
@@ -51,17 +51,16 @@ namespace libsemigroups {
 
     //! A constructor.
     //!
-    //! The parameter \p blocks must have length \f$n\f$ for some integer
-    //! \f$n > 0\f$, consist of non-negative integers, and have the property
-    //! that if \f$i\f$, \f$i > 0\f$ occurs in \p blocks, then \f$i - 1\f$
-    //! occurs earlier in \p blocks. None of this is checked.  The parameter
-    //! \p blocks is not copied, and is deleted by the destructor
-    //! Blocks::~Blocks.
-    //!
-    //! The parameter \p lookup must have length equal to the maximum value in
-    //! \p blocks, this maximum is also the number of blocks in the partition. A
-    //! value \c true in position \f$i\f$ indicates that the \f$i\f$th block is
-    //! signed! and \c false that it is unsigned.
+    //! The parameter \p blocks must be non-empty, consist of non-negative
+    //! integers, and have the property that if some positive \f$i\f$ occurs in
+    //! \p blocks, then \f$i - 1\f$ occurs earlier in \p blocks. None of this is
+    //! checked.  The parameter \p blocks is not copied, and is deleted by the
+    //! destructor Blocks::~Blocks.
+    //!
+    //! The parameter \p lookup must have length equal to the number of
+    //! different values in \p blocks (this is equal to the number of blocks in
```

Comment:
Maybe add that the number of different values in \p blocks is also the maximum value in \p blocks, to retain both bits of information,.